operations/helm/charts/mimir-distributed/templates/metamonitoring/_helpers.tpl

```diff
@@ -56,3 +56,15 @@
     cluster: {{ include "mimir.clusterName" $.ctx | quote}}
 {{- end -}}
 {{- end -}}
+
+{{- define "mimir.metaMonitoring.metrics.remoteReadUrl" -}}
+{{- with $.ctx.Values.metaMonitoring.grafanaAgent.metrics }}
+{{- $writeBackToMimir := not (.remote).url -}}
+{{- if $writeBackToMimir -}}
+{{- include "mimir.remoteReadUrl.inCluster" $.ctx }}
+{{- else -}}
+{{- $parsed := urlParse (.remote).url -}}
+{{ $parsed.scheme }}://{{ $parsed.host }}/prometheus
```

Comment:
@dimitarvdimitrov how do you feel about introducing a global field to the KEDA remote read URL? In some scenarios, people might leverage a custom deployed Prometheus (or Prometheus operator) for monitoring their mimir setup.

I am thinking about something like `kedaRemoteReadURL`, if unset we are using the metamonitoring URL (`mimir.remoteReadUrl.inCluster`), if set the one provided through the user.

Reply:
I think that makes sense. It's also some part of the follow-up work issue https://github.com/grafana/mimir/issues/7368. But my comment here was about not parsing the URL of metamonitoring and instead use it verbatim (`{{- (.remote).url -}}`), which still applies even if we add `kedaRemoteReadURL`


#### regarding `kedaRemoteReadURL`

Where would that go? With the current model it would fit under each component

```yaml
distributor:
  kedaAutoscaling:
    kedaRemoteReadURL: http://...

ruler:
  kedaAutoscaling:
    kedaRemoteReadURL: http://...

querier:
  kedaAutoscaling:
    kedaRemoteReadURL: http://...

query_frontend:
  kedaAutoscaling:
    kedaRemoteReadURL: http://...
```


It's less repetitive and slightly more maintainable if we have a single `kedaAutoscaling` section top-level in the values.yaml with all the common parameters:

```yaml
  kedaAutoscaling:
    prometheusAddress: http://... # I didn't like kedaKemoteReadURL because keda is repaeated and the URL isn't a remote-read per-se, it's a prometheus URL, which supports regular reads AND remote-read
    customHeaders:
      {}
    pollingInterval: 10
```

WDYT? I'm also slightly inclined to do this change in a follow-up PR. It will be a breaking change, but that's ok since this is an experimental feature.

Reply:
I added a comment to #7368 regarding that topic. I agree on adding this option in another PR. I will for now just go a head and not parse anything here.

Reply:
@dimitarvdimitrov I pushed my current working state. Regarding this parsing, I am still unsure what to do here. Sorry if I misunderstand you:

Using `{{- (.remote).url -}}` would result in `https://mimir.example.com/api/v1/push` which I think is not the correct Read URL.

Reply:
that's a really good point. The metamonitoring endpoint should end in `/push`, so reusing that directly isn't going to work. 

In this case parsing the URL as you had it probably makes the most sense, but it's rather limiting (wouldn't work with vanila Prometheus because that serves queries under `/api/v1/query`, not `/prometheus/api/v1/query`). I think as a first iteration it's ok to merge as-is and clarify this in the doc string above `kedaAutoscaling`. 

We can iterate on it as part of https://github.com/grafana/mimir/issues/7368

Reply:
Alright, thanks for the discussion!